### DISCUSSION TITLE
Example _currentTodo, is the throw UnimplementedError needed?

### DISCUSSION THREAD

**Discussion by pontino** [rrousselGit/riverpod#1214] [Q&A] [answered]
https://github.com/rrousselGit/river_pod/blob/eaa5a41e0de0be972c686a8e45914a1931d987ab/examples/todos/lib/main.dart#L232

Why do we need a Provider with an unimplemented error?

**Comment by maham** [comment, marked as answer]
The _currentTodo is used to specify which todo is currently being shown as a TodoItem. It's not really supposed to be used without first setting the item to show.
[Here](https://github.com/rrousselGit/river_pod/blob/eaa5a41e0de0be972c686a8e45914a1931d987ab/examples/todos/lib/main.dart#L121) the provider is overridden in a nested ProviderScope when a todo is selected. So the UnimplementedError is there as a guard against showing a TodoItem without first setting _which_ item to show.

A short answer but maybe something to help you find the pattern.

**Reply by pontino** [reply]
yes it does make sense, ty!
